Explorer/Assets/Plugins/RustSegment/rust-segment/src/operations.rs

```diff
@@ -23,21 +22,53 @@ pub fn new_track(
     })
 }
 
-pub fn new_identify(used_id: &str, traits_json: &str, context_json: &str) -> Option<Identify> {
+pub fn new_identify(user: User, traits_json: &str, context_json: &str) -> Option<Identify> {
     let traits_json: Value = as_option(serde_json::from_str(traits_json))?;
     let context_json: Value = as_option(serde_json::from_str(context_json))?;
 
     Some(Identify {
-        user: User::UserId {
-            user_id: used_id.to_string(),
-        },
+        user,
         traits: traits_json,
         context: Some(context_json),
         timestamp: Some(OffsetDateTime::now_utc()),
         ..Default::default()
     })
 }
 
+pub unsafe fn user_from(used_id: *const c_char, anon_id: *const c_char) -> Option<User> {
```

Comment:
should this parameter be called `user_id` instead?

Reply:
Shouldn't, it's Rust naming convention when you convert one type to another